python/CLOUDFRONT_VIEWER_POLICY_HTTPS/CLOUDFRONT_VIEWER_POLICY_HTTPS.py

```diff
@@ -0,0 +1,367 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+"""
+Description:
+  Checks whether your Amazon CloudFront Distributions use HTTPS (directly or via a redirection). The rule returns NON_COMPLIANT if the ViewerProtocolPolicy is set to 'allow-all' (i.e Viewers can use HTTP or HTTPS).
+Trigger:
+  Configuration Change on AWS::CloudFront::Distribution
+Reports on:
+  AWS::CloudFront::Distribution
+Rule Parameters:
+  None
+Scenarios:
+  Scenario: 1
+     Given: The value for 'ViewerProtocolPolicy' is 'allow-all' for at least one CacheBehavior
+      Then: Return NON_COMPLIANT
+  Scenario: 2
+     Given: The value for 'ViewerProtocolPolicy' is not 'allow-all' for all the CacheBehaviors
+      Then: Return COMPLIANT
+"""
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::CloudFront::Distribution'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    distribution_id = configuration_item['resourceId']
+    if configuration_item['configuration']['distributionConfig']['defaultCacheBehavior']['viewerProtocolPolicy'] == 'allow-all':
+        return build_evaluation(distribution_id, 'NON_COMPLIANT', event, DEFAULT_RESOURCE_TYPE, annotation='''Default ViewerProtocolPolicy is set to 'allow-all' for this Amazon CloudFront distribution.''')

```

Comment:
You need to use build_evaluation_from_config_item() as this is a configuration change rule.